### PR TITLE
Add 3-layer architect

### PR DESCRIPTION
**Pull request by ArtyomBanchikov**

### CODE REVIEW

File: Pets.API/Controllers/PetController.cs [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ﻿using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
-using Pets.API.Contexts;
-using Pets.API.Entities;
+using Pets.DAL.Contexts;
+using Pets.DAL.Entities;
```
Comment (hromau) [inline Pets.API/Controllers/PetController.cs]: api shouldn't know about dal layer, isn't it?

File: Pets.DAL/Interfaces/IPetRepository.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pets.DAL.Interfaces
+{
+    public interface IPetRepository<Pet>
```
Comment (hromau) [inline Pets.DAL/Interfaces/IPetRepository.cs]: why u use here generic type? is ur repo generic ?

**Review by hromau — CHANGES_REQUESTED** [review]

File: Pets.API/Controllers/PetController.cs (hunk 1 above)
Comment (ArtyomBanchikov) [reply Pets.API/Controllers/PetController.cs]: This is temporary to avoid mistakes in project

File: Pets.API/Controllers/PetController.cs [hunk 3]
```diff
@@ -18,13 +18,13 @@ public PetController(PetsContext context)
         }
 
         [HttpGet]
-        public IEnumerable<Pet> GetAll()
+        public IEnumerable<PetEntity> GetAll()
```
Comment (stas1408123) [inline Pets.API/Controllers/PetController.cs]: Don't touch Controller now

**Review by stas1408123 — CHANGES_REQUESTED** [review]

File: Pets.API/Controllers/PetController.cs (hunk 3 above)
Comment (ArtyomBanchikov) [reply Pets.API/Controllers/PetController.cs]: It automatically renamed in all files

Comment (stas1408123) [inline Pets.API/Controllers/PetController.cs]: PetViewModel

Comment (stas1408123) [reply Pets.API/Controllers/PetController.cs]: Add folder ViewModels

File: Pets.BLL/Services/PetService.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        IPetRepository Repository { get; set; }
+
+        public PetService(IPetRepository repository)
+        {
+            Repository = repository;
+        }
+
+        public IEnumerable<Pet> GetAll()
+        {
+            var mapper = new MapperConfiguration(cfg => cfg.CreateMap<PetEntity, Pet>()).CreateMapper();
```
Comment (stas1408123) [inline Pets.BLL/Services/PetService.cs]: Mayby should put mapper in constructor?
What do you think about if?

**Review by stas1408123 — CHANGES_REQUESTED** [review]

**Comment by stas1408123** [comment]
It is working?)

File: Pets.API/Controllers/PetController.cs [hunk 5]
```diff
@@ -1,60 +1,55 @@
...
+            Service = service;
+            _mapper = mapper;
         }
 
         [HttpGet]
-        public IEnumerable<Pet> GetAll()
+        public IEnumerable<PetViewModel> GetAll()
         {
-            return _context.Pets.ToList();
+            return _mapper.Map<IEnumerable<Pet>, IEnumerable<PetViewModel>>(Service.GetAll());
```
Comment (hromau) [inline Pets.API/Controllers/PetController.cs]: looks like spaghetti-code. try to split it a little bit

File: Pets.BLL/Services/PetService.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+using Pets.DAL.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pets.BLL.Services
+{
+    internal class PetService: IPetService
```
Comment (hromau) [inline Pets.BLL/Services/PetService.cs]: why internal?

File: Pets.BLL/Services/PetService.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        IPetRepository Repository { get; set; }
```
Comment (hromau) [inline Pets.BLL/Services/PetService.cs]: why not private?
and readonly?

Comment (hromau) [inline Pets.BLL/Services/PetService.cs]: and it should be a Field

**Review by hromau — CHANGES_REQUESTED** [review]

File: Pets.BLL/Services/PetService.cs (hunk 6 above)
Comment (ArtyomBanchikov) [reply Pets.BLL/Services/PetService.cs]: forgot to change

**Review by stas1408123 — APPROVED** [review]

File: Pets.API/Controllers/PetController.cs [hunk 8]
```diff
@@ -1,60 +1,57 @@
...
         }
 
         [HttpGet]
-        public IEnumerable<Pet> GetAll()
+        public IEnumerable<PetViewModel> GetAll()
         {
-            return _context.Pets.ToList();
+            return _mapper
+                .Map<IEnumerable<Pet>, IEnumerable<PetViewModel>>(_service
+                .GetAll());
```
Comment (hromau) [inline Pets.API/Controllers/PetController.cs]: kek))) 
var allPets = _service.GetAll(); return _mapper.Map ......

**Review by hromau — CHANGES_REQUESTED** [review]

File: Pets.API/Controllers/PetController.cs [hunk 9]
```diff
@@ -1,60 +1,57 @@
...
+            _mapper = mapper;
         }
 
         [HttpGet]
-        public IEnumerable<Pet> GetAll()
+        public IEnumerable<PetViewModel> GetAll()
         {
-            return _context.Pets.ToList();
+            return _mapper
+                .Map<IEnumerable<Pet>, IEnumerable<PetViewModel>>(_service
```
Comment (hromau) [inline Pets.API/Controllers/PetController.cs]: i mean var allPets = service.GetAll(); return mapper.MAp().......

**Review by hromau — CHANGES_REQUESTED** [review]

**Review by hromau — APPROVED** [review]

**Pull request merged**